analyzers/nodejs/integration_test.go

```
@@ -102,59 +93,46 @@ func assertProjectFixtureExists(t *testing.T, name string) {
 	assert.True(t, exists, name+" did not have its node modules installed")
 }
 
-func initializeProjects(testDir string) error {
-	var waitGroup sync.WaitGroup
-	waitGroup.Add(len(projects))
-
-	for _, project := range projects {
-		go func(proj fixtures.Project) {
-			defer waitGroup.Done()
-
-			projectDir := filepath.Join(testDir, proj.Name)
-			log.Debug("initializing " + projectDir)
-			nodeModulesExist, err := files.ExistsFolder(projectDir, "node_modules")
-			if err != nil {
-				panic(err)
-			}
+func projectInitializer(proj fixtures.Project, projectDir string) error {
+	nodeModulesExist, err := files.ExistsFolder(projectDir, "node_modules")
+	if err != nil {
+		return err
+	}
 
-			if nodeModulesExist {
-				log.Debug("node modules already exists for " + proj.Name + "skipping initialization")
-				return
-			}
+	if nodeModulesExist {
+		log.Debug("node modules already exists for " + proj.Name + "skipping initialization")
+		return nil
+	}
 
-			_, errOut, err := exec.Run(exec.Cmd{
-				Name:    "npm",
-				Argv:    []string{"install", "--production"},
-				Dir:     projectDir,
-				WithEnv: proj.Env,
-				Command: "npm",
-			})
-			if err != nil {
-				log.Error(errOut)
-				log.Error("failed to run npm install on " + proj.Name)
-			}
+	_, errOut, err := exec.Run(exec.Cmd{
+		Name:    "npm",
+		Argv:    []string{"install", "--production"},
+		Dir:     projectDir,
+		WithEnv: proj.Env,
+		Command: "npm",
+	})
+	if err != nil {
+		log.Error(errOut)
+		log.Error("failed to run npm install on " + proj.Name)
```

We should return this error too.